File: Algorithms/searchingAndTraversal/LinearSearch.java
package searchingAndTraversal;

import java.util.Scanner;

public class LinearSearch {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		
		
		int arr[] , n, searchValue;

	    System.out.print("Enter the number of elements: ");
	    n = in.nextInt();
	    arr = new int[n];

	    System.out.printf("Enter %d elements:\n", n);
	    for(int i=0; i<n; i++)
	        arr[i] = in.nextInt();

	    System.out.print("Enter the search value: ");
	    searchValue = in.nextInt();

	    linearSearch(arr, n, searchValue);
	    
	    in.close();
	}
	
	public static void linearSearch(int[] arr, int n, int searchValue) {
	    for(int i=0; i<n; i++) {
	        if (arr[i] == searchValue) {
	        	System.out.printf("%d is present at location %d.\n", arr[i], i+1);
	            return;
	        }
	    }
	    System.out.printf("%d isn't present in the array.\n", searchValue);
	}
}

File: Algorithms/searchingAndTraversal/LinearSearch.cpp
#include<bits/stdc++.h>
using namespace std;

void linearSearch(int arr[], int n, int searchValue) {
    for(int i=0; i<n; i++) {
        if (arr[i] == searchValue) {
            printf("%d is present at location %d.\n", arr[i], i+1);
            return;
        }
    }
    printf("%d isn't present in the array.\n", searchValue);
}


int main()
{
    int arr[100], n, searchValue;

    printf("Enter the number of elements: ");
    scanf("%d", &n);

    printf("Enter %d elements:\n", n);
    for(int i=0; i<n; i++)
        scanf("%d", &arr[i]);

    printf("Enter the search value: ");
    scanf("%d", &searchValue);

    linearSearch(arr, n, searchValue);

    return 0;
}
